app/observers/update_router_observer.rb

```diff
@@ -5,9 +5,13 @@ def after_save(artefact)
     RoutableArtefact.new(artefact).submit if artefact.live?
 
     unless artefact.owning_app == 'whitehall'
-      # Relying on current behaviour where this does not raise errors
-      # if done more than once, or done on artefacts never put live
-      RoutableArtefact.new(artefact).delete if artefact.archived?
+      if artefact.archived?
+        if artefact.redirect_url.blank?
+          RoutableArtefact.new(artefact).delete
+        else
+          RoutableArtefact.new(artefact).redirect(artefact.redirect_url)
+        end
+      end
```

Comment:
This feels like the wrong place to have all this logic (it was already a bit of a smell with the 2 cases.  now that we're adding a 3rd, I think it's the right time to improve it).  I think it would be better to move this inside `RoutableArtefact.submit` so that this observer merely calls `submit` and doesn't have to know anything about artefact states. `RouteableArtefact` can then do the right thing based on the state of the artefact etc.